### PR TITLE
Feature(Next-Gen Dataset): Integrate new stratified patching into the dataset

### LINKED ISSUE #715: Next-Gen Dataset: Integrating stratifed patching into NG Dataset
## Problem

Once #710 is merged, the new patching strategy should be integrated to the NG Dataset.

### PR DESCRIPTION
**Pull request by melisande-c**
## Description

> [!NOTE]  
> **tldr**: This PR makes it possible to use the new stratified patching strategy in the next-generation dataset.

### Background - why do we need this PR?

A stratified patching strategy was introduced in #710 with 2 applications in mind: validation data splitting, and more efficient masking. This PR allows it to be used by the `CAREamicsDataset`.

### Overview - what changed?

A new `StratifiedPatchingConfig` pydantic class, and some minor changes to the `NGDataConfig`.

### Implementation - how did you implement the changes?

Simply added a new case to the match-case block in the `create_patching_strategy` factory. And in the `NGDataConfig` validation "stratified" is a valid patching strategy choice for validation

### New features or files

- `StratifiedPatchingConfig`

### Modified features or files

- `NGDataConfig`
- `create_patching_strategy`

## How has this been tested?

Currently working on notebooks to benchmark. Denoising results so far look good but stratified patching seems to be a little slower. I will include results here shortly.

EDIT: performance benchmarking

Could be a bit more rigorous but the set-up is:
- Machine: M3 Mac with 16GB Memory:
- Data: SEM (long scan time as GT for PSNR)
- Training: 10 epochs (5 repeats)

Full train time:
- Stratified: 40.9 ± 0.9 s
- Random: 40.0 ± 1.4 s

PSNR:
- Stratified: 18.9 ± 0.2
- Random: 18.5 ± 0.4

## Related Issues

- Resolves #715 

## Additional Notes and Examples

I haven't changed any of the convenience functions, so currently the easiest way to test with the new configuration is to do something like:

```python
from careamics.config.ng_factories import create_n2v_configuration
from careamics.config.data.patching_strategies import StratifiedPatchingConfig

config = create_n2v_configuration(
    experiment_name="example",
    data_type="array",
    axes="YX",
    patch_size=(64, 64),
    batch_size=32,
    num_epochs=50,
    use_n2v2=False,
    train_dataloader_params={"num_workers": 0}
)
config.data_config.patching = StratifiedPatchingConfig(patch_size=(64, 64))
```

---

**Please ensure your PR meets the following requirements:**

- [x] Code builds and passes tests locally, including doctests
- [ ] New tests have been added (for bug fixes/features)
- [x] Pre-commit passes
- [ ] PR to the documentation exists (for bug fixes / features)

### CODE REVIEW

File: src/careamics/config/data/patching_strategies/__init__.py [hunk 1]
```diff
@@ -4,12 +4,16 @@
...
     "RandomPatchingConfig",
     "SequentialPatchingConfig",
+    "StratifiedPatchingConfig",
     "TiledPatchingConfig",
     "WholePatchingConfig",
+    "_PatchedConfig",
 ]
 
 
+from ._patched_config import _PatchedConfig
```
Comment (jdeschamps) [inline src/careamics/config/data/patching_strategies/__init__.py]: Minor point. Should we make it not private? It feels strange to import private classes in an `__init__`.

File: src/careamics/config/data/ng_data_config.py [hunk 2]
```diff
@@ -881,7 +886,7 @@ def convert_mode(
                     patch_size=list(new_patch_size), overlaps=list(overlap_size)
                 )
         else:  # validating
-            assert isinstance(self.patching, RandomPatchingConfig)  # for mypy
+            assert isinstance(self.patching, _PatchedConfig)  # for mypy
```
Comment (jdeschamps) [inline src/careamics/config/data/ng_data_config.py]: Tiling is also an instance of `_PatchedConfig`

**Review by jdeschamps — APPROVED** [review]
Only one comment that may be an issue, but otherwise approving.

File: src/careamics/config/data/patching_strategies/__init__.py (hunk 1 above)
Comment (melisande-c) [reply src/careamics/config/data/patching_strategies/__init__.py]: Removed from import

File: src/careamics/config/data/ng_data_config.py (hunk 2 above)
Comment (melisande-c) [reply src/careamics/config/data/ng_data_config.py]: checking is not WholePatchingStrategy instead

**Review by jdeschamps — APPROVED** [review]

File: src/careamics/config/data/ng_data_config.py (hunk 2 above)
Comment (jdeschamps) [reply src/careamics/config/data/ng_data_config.py]: Makes sense!
